### ISSUE TITLE
Database syncing

### ISSUE THREAD

**Issue by miller9904** [skylight-software/impulse#7]
Now that we're working on a database schema (#6), we can also start work on the database syncing method.  I believe that [this project](https://github.com/matthew-andrews/offline-todo-with-sync) has already implemented a database syncing method of some kind.  Maybe we could use this code as is, or base our implementation off of it.  We will be using [WebRTC](https://developer.mozilla.org/en-US/docs/Web/API/WebRTC_API) to sync the data, of course.

**Comment by elogent** [comment]
In order to make the syncing as smooth and easy for the user as possible, we should write our own logic that is designed around our specific database. From what we've discussed so far about the schema, it seems like a simple exchange could work:

*   Device A and Device B sync with each other
*   Any data that has expired on either A or B is immediately deleted
*   Any data that is currently on A but not on B is copied from A to B, and vice versa

Note that this doesn't take into account updates to existing tuples in the database or explicit deletion of data not based on expiration.

**Comment by miller9904** [comment]
OK, that sounds good.  I just thought of something, though.  To sync, Impulse needs at least two instances open at the same time.  I don't think most people leave web pages open much.  And plus, most computers go to sleep when the lid is shut.  So how are we going to solve that problem?

**Comment by elogent** [comment]
iTunes requires two devices to be running Apple's software at the same point in time in order to sync, so I don't see how it wouldn't work for Impulse as well.

**Comment by miller9904** [comment]
I don't use Apple devices much, but if the software is native, then it can always run in the background.  Web apps can't do background sync yet, and it doesn't seem right to gamble on the fact that the user will leave Impulse open all the time.

**Comment by elogent** [comment]
True. I'd say the best solution in this case would be to make sync initiation manual.
